packages/server/src/trpc-routers/__tests_e2e__/swap-router.test.ts

```diff
@@ -0,0 +1,223 @@
+import { CoinPretty, Int, PricePretty, RatePretty } from "@keplr-wallet/unit";
+import { isNumeric } from "@osmosis-labs/utils";
+import { initTRPC } from "@trpc/server";
+
+import { AssetLists } from "../../queries/__tests__/mock-asset-lists";
+import { MockChains } from "../../queries/__tests__/mock-chains";
+import { createInnerTRPCContext } from "../../trpc";
+import { superjson } from "../../utils/superjson";
+import { swapRouter } from "../swap-router";
+
+const { createCallerFactory, router: createTRPCRouter } = initTRPC
+  .context<typeof createInnerTRPCContext>()
+  .create({
+    transformer: superjson,
+  });
+const createCaller = createCallerFactory(createTRPCRouter({ swapRouter }));
+const caller = createCaller({
+  assetLists: AssetLists,
+  chainList: MockChains,
+});
+
+const percentageRegex = /^(<\s)?\d+\.\d+%$/;
+const atomMinimalDenom =
```

Comment:
For my understanding, what does "minimal" stand for?

Reply:
`minimalDenom` represents the base IBC hash denom in the Osmosis chain.

Reply:
This is outside of the scope of this PR. I know that the naming convention has been here before but would it make sense to standardize it with chain?

In SQS and chain, we are calling them as "ibcDenom"s. I'm noticing that the mobile is inheriting the FE naming convention.

I think that standardization across the stack would be helpful long term

Reply:
Thank you. Long term, it does make sense to ensure consistency across our entire stack. We, and maybe mobile too, chose this naming convention to maintain compatibility with Keplr's `CoinPretty` object, which follows the same definition and property name

A practical first step would be to create a dictionary document that outlines all terminology. This will help prevent similar situations from happening in the future. Following that, we'll need to devise a strategy for renaming these objects, a process that will likely require updates to our primitive objects.